# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.h
# Commit: d77b78a71a7e
# Full Hash: d77b78a71a7e76f975052eee5c2fef7baeb88d44
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-20 21:21:03
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - Support conversation between InternalRequest and IPCInternalRequest. r=dom-worker-reviewers,jesup
#   
#   Depends on D138812
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138813
# ==============================================================================

diff -r dcfa6d7e82c0 -r d77b78a71a7e dom/fetch/InternalRequest.h
--- a/dom/fetch/InternalRequest.h	Fri Jan 20 09:09:16 2023 +0000
+++ b/dom/fetch/InternalRequest.h	Fri Jan 20 09:09:17 2023 +0000
@@ -29,6 +29,7 @@
 namespace mozilla {
 
 namespace ipc {
+class PBackgroundChild;
 class PrincipalInfo;
 }  // namespace ipc
 
@@ -97,6 +98,9 @@
 
   explicit InternalRequest(const IPCInternalRequest& aIPCRequest);
 
+  void ToIPCInternalRequest(IPCInternalRequest* aIPCRequest,
+                            mozilla::ipc::PBackgroundChild* aManager);
+
   SafeRefPtr<InternalRequest> Clone();
 
   void GetMethod(nsCString& aMethod) const { aMethod.Assign(mMethod); }